tview/application.go

```diff
@@ -67,17 +68,17 @@ type Application struct {
 	// stop the application.
 	screenReplacement chan tcell.Screen
 
-	// Defines whether a bracketed paste is currently ongoing.
-	pasteActive bool
+	VimMode	*vim.Vim
 }
 
 // NewApplication creates and returns a new application.
-func NewApplication() *Application {
+func NewApplication(vimEnabled bool) *Application {
```

Comment:
Why does tview need to know about vim mode?

Reply:
Application holds the pointer to vim.Vim that will be shared to the entire program, with app.vimMode.

Reply:
But this is a cordless thing, not a tview thing, right?

Reply:
It is a cordless thing, but the best place as it seemed to me to share the global pointer to vim.Vim was in window.app and derived. So NewApplication needs to know if vim is enabled (as it does not have access to the config)